packages/runtime/container-runtime/src/blobManager/blobManager.ts

```diff
@@ -335,6 +335,35 @@ export class BlobManager {
 		return this.redirectTable.get(blobId) !== undefined;
 	}
 
+	/**
+	 * Lookup the blob URL for a given local blob id.
+	 * @param localId - The local blob id. Likely coming from a handle.
+	 * @returns The blob URL if found and the blob is not pending, undefined otherwise.
+	 * @remarks
+	 * This leverages the driver's URL building logic to create a direct access URL for the blob.
+	 * For blobs with pending payloads (localId exists but upload hasn't finished), this returns undefined. 
+	 * Consumers should use the observability APIs on the handle (handle.payloadState, payloadShared event) 
+	 * to understand/wait for URL availability.
+	 * 
+	 * **WARNING**: The returned URL may expire and does not support permalinks.
+	 * This is intended for temporary integration scenarios only.
+	 */
+	public lookupBlobURL(localId: string): string | undefined {
```

Comment:
The URL concept should not be plumbed through the runtime, but rather the runtime should offer the storageId back out (this is already a plumbed concept from the driver's createBlob).

The driver (or URL resolver?) _may_ then offer some API to get a URL from that storage id (here meaning a URL that will return the blob if an HTTP GET request is made to it).  Bearing in mind that not all drivers/services support URL addressability, particularly local-driver.

getAbsoluteUrl is another good example of where plumbing the URL through has been painful for us and we don't recommend it anymore.